contracts/test/SimplePDPService.t.sol

```diff
@@ -93,85 +93,152 @@ contract SimplePDPServiceTest is Test {
 contract SimplePDPServiceFaultsTest is Test {
     SimplePDPService public pdpService;
     address public pdpVerifierAddress;
+    uint256 public proofSetId;
+    uint256 public leafCount;
+    uint256 public seed;
+    uint256 public challengeCount;
 
     function setUp() public {
         pdpVerifierAddress = address(this);
         SimplePDPService pdpServiceImpl = new SimplePDPService();
         bytes memory initializeData = abi.encodeWithSelector(SimplePDPService.initialize.selector, address(pdpVerifierAddress));
         MyERC1967Proxy pdpServiceProxy = new MyERC1967Proxy(address(pdpServiceImpl), initializeData);
         pdpService = SimplePDPService(address(pdpServiceProxy));
+        proofSetId = 1;
+        leafCount = 100;
+        seed = 12345;
+        challengeCount = 5;
     }
 
     function testPosessionProvenOnTime() public {
-        uint256 proofSetId = 1;
-        uint256 challengedLeafCount = 100;
-        uint256 seed = 12345;
-        uint256 challengeCount = 5;
-
         // Set up the proving deadline
         pdpService.rootsAdded(proofSetId, 0, new PDPVerifier.RootData[](0));
-        // Warp to just before the deadline
-        vm.warp(block.number + pdpService.getMaxProvingPeriod() - 1);
-        pdpService.posessionProven(proofSetId, challengedLeafCount, seed, challengeCount);
+        vm.roll(block.number + pdpService.getMaxProvingPeriod() - 1);
+        pdpService.posessionProven(proofSetId, leafCount, seed, challengeCount);
         assertTrue(pdpService.provenThisPeriod(proofSetId));
     }
 
-    function testPosessionProvenLate() public {
-        uint256 proofSetId = 1;
-        uint256 challengedLeafCount = 100;
-        uint256 seed = 12345;
-        uint256 challengeCount = 5;
+    function testNextProvingPeriodCalledLastMinuteOK() public {
+        pdpService.rootsAdded(proofSetId, 0, new PDPVerifier.RootData[](0));
+        vm.roll(block.number + pdpService.getMaxProvingPeriod() - 1);
```

Comment:
By my reading, this should also succeed one epoch later (and same below).

Could you add a test for the first epoch at which the proof is too late?

Reply:
Thanks this helped me find a bug